### DISCUSSION TITLE
Enable to use `temurin` as a Java distribution name instead of `adoptopenjdk`  in asdf .tool-version

### DISCUSSION THREAD

**Discussion by kota65535** [renovatebot/renovate#24927] [Suggest an Idea]
### What would you like Renovate to be able to do?

Now AdoptOpenJDK is moved to [Eclipse Adoptium project](https://blog.adoptopenjdk.net/2021/03/transition-to-eclipse-an-update/) and also renamed to Temurin.
Since asdf-java plugin already handles these Temurin versions like `temurin-17.0.5+8`, so Renovate should accept `temurin` as well as `adoptopenjdk` as a distribution name.

### If you have any ideas on how this should be implemented, please tell us here.

Maybe add a pattern to this regex below
https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/asdf/upgradeable-tooling.ts#L192

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by HonkingGoose** [comment]
Are we sure we only need to change the regex linked above? Maybe we should change more things like:

1. Do we want to add a new _replacement_ preset to replace `adoptopenjdk` with `temurin`, file location: https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/replacements.ts
1. Our code and documentation still uses terms like:
    - `AdoptOpenJDK`
    - `adoptium-java`
    - `AdoptiumJavaResponse`
  Basically: search for `adopt` and see what pops up. 😉 Do we want to change those terms?
1. We should maybe update the code for `javaLTSVersions` as that refers to `adoptopenjdk`: https://github.com/renovatebot/renovate/blob/384a235363ad0a6b984cd76961b4f5de3b51e6c1/lib/config/presets/internal/workarounds.ts#L123-L144
1. Maybe rename `adoptium-java`/`AdoptiumJavaDatasource` datasource?`
